theano/tensor/opt.py

```diff
@@ -63,6 +63,54 @@
 # Utilities
 
 
+def copy_stack_trace(from_var, to_var):
+    """
+    Copies the stack trace from one or more tensor variables to
+    one or more tensor variables.
+
+    :param from_var: tensor variable or list of tensor variables to
+                     copy stack traces from.
+    :param to_var: tensor variable or list of tensor variables to
+                     copy stack traces to.
+
+    .. note:: The stacktrace is assumed to be of the form of a list of lists
+    of tuples. Each tuple contains the filename, line number, function name
+    and so on. Each list of tuples contains the truples belonging to a
+    particular variable.
+    """
+
+    # Store stack traces from from_var
+    tr = []
+    if type(from_var) is list:
+        # If from_var is a list, store concatenated stack traces
+        if len(from_var) > 0:
+            for v in from_var:
+                if hasattr(v.tag, 'trace') and len(v.tag.trace) > 0:
```

Comment:
The second condition is not useful. A list + an empty list, return the original list.